contracts/fair-launch/src/exchange_actions.rs

```diff
@@ -0,0 +1,296 @@
+                    removed = true;
+                    current_endpoints.remove(i);
+
+                    break;
+                }
+            }
+
+            require!(removed, "Unknown endpoint name");
+        }
+    }
+
+    /// forward an execute on dest context call on an exchange SC
+    #[payable("*")]
+    #[endpoint(forwardExecuteOnDest)]
+    fn forward_execute_on_dest(
+        &self,
+        dest: ManagedAddress,
+        endpoint_name: ManagedBuffer,
+        extra_args: MultiValueEncoded<ManagedBuffer>,
+    ) {
+        self.require_not_paused();
+        self.require_not_initial_launch();
+
+        let egld_value = self.call_value().egld_value().clone_value();
+        require!(egld_value == 0, "Invalid payment");
+
+        let caller = self.blockchain().get_caller();
+        let payments = self.call_value().all_esdt_transfers().clone_value();
+        let endpoint_info = self.find_endpoint_info(&dest, &endpoint_name);
+
+        let mut input_fees_percentage = ManagedVec::new();
+        for _ in 0..payments.len() {
+            input_fees_percentage.push(endpoint_info.input_fee_percentage);
+        }
+
+        let (_, back_transfers) = if !payments.is_empty() {
+            let take_fees_result =
+                self.take_fees(caller.clone(), payments.clone(), input_fees_percentage);
+
+            if endpoint_info.burn_input {
+                self.burn_all_tokens(&take_fees_result.fees);
+            }
+
+            ContractCallNoPayment::<_, MultiValueEncoded<ManagedBuffer>>::new(dest, endpoint_name)
+                .with_multi_token_transfer(take_fees_result.transfers)
+                .with_raw_arguments(ManagedArgBuffer::from(extra_args.into_vec_of_buffers()))
+                .execute_on_dest_context_with_back_transfers::<MultiValueEncoded<ManagedBuffer>>()
+        } else {
+            ContractCallNoPayment::<_, MultiValueEncoded<ManagedBuffer>>::new(dest, endpoint_name)
+                .with_raw_arguments(ManagedArgBuffer::from(extra_args.into_vec_of_buffers()))
+                .execute_on_dest_context_with_back_transfers::<MultiValueEncoded<ManagedBuffer>>()
+        };
+
+        if !back_transfers.esdt_payments.is_empty() {
+            let mut output_fees_percentage = ManagedVec::new();
+            for _ in 0..back_transfers.esdt_payments.len() {
+                output_fees_percentage.push(endpoint_info.output_fee_percentage);
+            }
+
+            let take_fees_from_results =
+                self.take_fees(caller, back_transfers.esdt_payments, output_fees_percentage);
+
+            if endpoint_info.burn_output {
+                self.burn_all_tokens(&take_fees_from_results.fees);
+            }
+
+            self.send().direct_multi(
+                &take_fees_from_results.original_caller,
+                &take_fees_from_results.transfers,
+            );
+        }
+    }
+
+    // Disabled until a solution is found, no way of knowing the results for now.
```

Comment:
delete this - as we will need asyncV3 and backTransfers on async. Until then there is no need for this to stay here.

Reply:
Done.